js/app/file-list-page/file-list-page/index.js

```
@@ -71,24 +72,32 @@ export class FileListPage extends Component {
    * @private
    */
   _initNestedComponentsWithDataFromServer() {
-    this._apiService.userPreferences()
-        .then((userPreferences) => {
-          this._userDetails.username = userPreferences.name;
+    this._stateService.dispatch(actionNames.getUserDetails).then(() => {
+      this._stateService.dispatch(actionNames.loadCurrentFolder).catch(() => {
+      });
 
-          this._apiService.getFolder(userPreferences.rootFolderId).then((folder) => {
-            this._breadcrumbs.addBreadcrumb(folder.name);
-          }).catch((error) => {
-            this._breadcrumbs.errorMessage = error.message;
-          });
+      this._stateService.dispatch(actionNames.loadFileList).catch(() => {
+      });
+    }).catch(() => {
+    });
```

Why?

---

Delete.

---

Because I need to throw an error if userDetails did not load, otherwise there will be a request for the following components.

---

Done.